router/handler_link/bind.go

```diff
@@ -166,11 +164,19 @@ func (self *heartbeatCallback) HeartbeatRespRx(ts int64) {
 func (self *heartbeatCallback) CheckHeartBeat() {
 	log := pfxlog.Logger().WithField("channelId", self.ch.Label())
 	now := time.Now().UnixMilli()
-	if self.firstSent != 0 && (now-self.firstSent > 30000) && (now-self.lastResponse > 30000) {
+	if delta := now - self.lastResponse; delta > 30000 {
 		log.Warn("heartbeat not received in time, link may be unhealthy")
 		self.latencyMetric.Clear()
 		self.latencyMetric.Update(88888888888)
```

Comment:
maybe `math.MaxUInt64`?

Reply:
It just needs to be sufficiently large and a marker value, so also easy to recognize when someone is looking at it. Before this logic was added we never updated the latency if no response was received, so we could be reporting low values. I'll add a couple more eights, which will bump it up to 2 hours.